plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/AppEngineDeployPreferencesPanel.java

```diff
@@ -223,12 +225,40 @@ private void setupProjectSelectorDataBinding() {
         gcpProjectToProjectId, projectIdToGcpProject);
   }
 
-  private void setupTextFieldDataBinding(Control control, String modelPropertyName,
+  /**
+   * Binds a {@link Text} field with a property of the {@link DeployPreferences deploy
```

Comment:
Why not make this `setupPossiblyUnvalidatedTextFieldDataBinding` or `OptionallyValidated`